color.js

```diff
@@ -12,11 +12,23 @@ export class Color {
   }
 
   //"Factory method" that creates color from given HEX value
-  static fromHex(hexColor) {
+  static hexToRGB(hexColor) {
+    const r = parseInt(hexColor.slice(1, 3), 16);
+    const g = parseInt(hexColor.slice(3, 5), 16);
+    const b = parseInt(hexColor.slice(5, 7), 16);
+    const res = { r, g, b };
+    return res;
   }
 
   //Converts color to HEX format
-  toHex() {
+  rgbToHex(rgbColor) {
+    let hex = '0x';
+    for (const key in rgbColor) {
+      hex += rgbColor[key].toString(16);
+      if (rgbColor[key] === 0) hex += '0';
+    }
+    const res = hex;
+    return res;
```

Comment:
You have problem not only with 0 number, but with all numbers less than 16 (1=>1, 15=>f, 10=>a) 
My suggestion:

function rgbToHex(rgbColor) {
  let hex = '#';
  for (const c of Object.values(rgbColor)) {
    let value = c.toString(16);
    if(value.length == 1) value = "0" + value;
    hex += value;
  }
  return hex;
}